crates/noirc_evaluator/src/ssa_refactor/ir/instruction.rs

```diff
@@ -237,7 +237,9 @@ impl Instruction {
         match self {
             Instruction::Binary(binary) => binary.simplify(dfg),
             Instruction::Cast(value, typ) => {
-                if let Some(value) = (*typ == dfg.type_of_value(*value)).then_some(*value) {
+                if let Some(constant) = dfg.get_numeric_constant(*value) {
+                    SimplifiedTo(dfg.make_constant(constant, typ.clone()))
+                } else if let Some(value) = (*typ == dfg.type_of_value(*value)).then_some(*value) {
```

Comment:
This is not needed but is a small change I saw was missing when seeing `v0 = cast Field 0 as u32` in the output.